joint_state_broadcaster/doc/userdoc.rst

```diff
@@ -38,3 +38,22 @@ An example parameter file
 
 .. generate_parameter_library_default::
   ../src/joint_state_broadcaster_parameters.yaml
+
+
+Order of the joints in the message
+----------------------------------
+
+The order of the joints in the message can determined by 3 different parameter settings:
+
+1. No defined ``joints`` parameter and ``use_urdf_to_filter`` set to ``false``:
+    The order of the joints in the message is the same as the order of the joints' state interfaces registered in the resource manager. This is typically the order in which the hardware components are loaded and configured.
+
+2. No defined ``joints`` parameter and ``use_urdf_to_filter`` set to ``true``:
+    The order of the joints in the message is the same as the order of the joints in the URDF file, which is inherited from the loaded URDF model and independent of the order in the `ros2_control` tag.
+
+3. Defined ``joints`` parameter along with ``interfaces`` parameter:
+    The order of the joints in the message is the same as the order of the joints in the ``joints`` parameter.
+
+    If the ``joints`` parameter is a subset of the total available joints in the URDF (or) the total available state interfaces, the order of the joints in the message is the same as the order of the joints in the ``joints`` parameter.
```

Comment:
isn't this a half-duplicate of line 55?

Reply:
Yes! It is

Don't you like it?

Reply:
if it is intended like this, then I don't understand it :D

Reply:
I'll fix it

Reply:
@christophfroehlich fixed it